chainer/functions/array/tile.py

```diff
@@ -15,22 +17,23 @@ def __init__(self, reps):
                 isinstance(x, six.integer_types) for x in reps):
             self.reps = reps
         else:
-            raise TypeError('reps must be int or tuple of ints')
+            msg = 'reps must be int or tuple of ints. \n' \
+                  'Actual: {0}'.format(type(reps))
+            raise TypeError(msg)
 
         if not all(x >= 0 for x in self.reps):
-            raise ValueError('all elements in reps must be zero or larger')
+            raise ValueError('All elements in reps must be zero or larger')
 
     def check_type_forward(self, in_types):
         type_check.expect(in_types.size() == 1)
 
     def forward(self, inputs):
-        self.retain_inputs(())
         self._in_shape = inputs[0].shape
         self._in_dtype = inputs[0].dtype
```

Comment:
This line seems unneeded.

Reply:
I think you commented on the old code. It has been removed!

Reply:
Ah, both old code and new code does not using `_in_dtype`.

Reply:
Thanks, I'll fix this one.